### ISSUE TITLE
Frient KEYZB-110

### ISSUE THREAD

**Issue by DaddyBabbes** [KiwiHC16/Abeille#2715]
Bonjour,
Je viens d’acquérir le clavier intelligent de Frient pour piloter mes scénarios d’alarme et je rencontre quelques soucis.
J’utilise une pizigate + plugin Abeille avec inclusions de plusieurs détecteurs d’ouvertures de porte, capteurs de température et d’humidité et 2 détecteurs de fumée Frient. Tous les composants fonctionnent correctement.
Lors de l’inclusion du clavier, il n’était pas reconnu et j’ai dût choisir le modèle présent dans la lsite
Une liste de commande à été configuré automatiquement (par le choix du modèle je présume)
Par contre, lorsque je lance le log en temps réel et que je tape une touche de fonction, rien ne remonte . J’ai aussi regardé la vidéo tuto de Quenton mais je ne retrouve pas les mêmes commandes.
Est ce que quelqu’un aurait réussi à faire fonctionner son Jeedom + PiZigate + Abeille + Clavier Firent ?
Je ne suis pas un pro de jeedom et loin de là mais et j’aurais aimé de l’aide pour diagnostiquer.
Je viens de passer en mode debug et relancé le démon. Je vais refaire quelques tests ce soir et récupérer les logs.
D'ici là je joins une copie d'écran des commandes qui sont remontées automatiquement suite au choix du modèle.
![Frient-KEYPAD-Cmd](https://github.com/KiwiHC16/Abeille/assets/16014749/a26c307c-db8d-43d2-8726-d42fa386c5d9)

**Comment by tcharp38** [comment]
Salut @DaddyBabbes 
J'ai besoin du package de logs que tu peux obtenir via la page maintenance + telecharger tout.
Et tu n'as qu'a glisser déposer ici ensuite

Merci

**Comment by DaddyBabbes** [comment]
Page maintenance = Analyse / Logs ?

**Comment by tcharp38** [comment]
Non, la page maintenance d'Abeille

![image](https://github.com/KiwiHC16/Abeille/assets/35221038/1be054f5-aa15-40e8-8dcc-cb63a08c3488)

**Comment by DaddyBabbes** [comment]
[AbeilleLogs-240624.tar.gz](https://github.com/user-attachments/files/15956229/AbeilleLogs-240624.tar.gz)
Merci, voici les logs téléchargé depuis la page Maintenance du plugin Abeille.

**Comment by tcharp38** [comment]
Je ne comprends déja pas pourquoi il n'a pas été reconnu.
Le modèle est fonctionnel normalement (voir #2525).

Peux tu par ex appuyer sur SOS, noter l heure, et me redonner le package de logs ?

Par ailleurs je te suggere de mettre à jour ta Zigate. Tu es en 3.21. La 3.23 est la derniere et ses corrections peuvent aider.

**Comment by DaddyBabbes** [comment]
OK, comme je ne suis pas devant le boitier, je ferai le test ce soir et je renverrai les logs.
Pour la mise à jour de la ZiGate, j'ai essayé depuis le plugin Abeille mais lorsque je clic sur Mettre à Jour et en ayant choisi 0004-0323 OPDM, il ne se passe rien.
J'ai essayé de trouver la procédure de mise à jour mais je ne trouve pas le binaire

**Comment by tcharp38** [comment]
Ha..
la doc officielle pour la mise à jour est la https://zigate.fr/documentation/

Mais via Abeille .. il faut que je creuse de mon coté avec ma Pi Zigate.

**Comment by DaddyBabbes** [comment]
Je n'ai pas pu mettre a jour la pizigate mais j'ai refais une inclusion (non reconnu et reconnu uniquement après réparation) et voici les logs full abeille.
J'espère que ca aidera.
[AbeilleLogs-240624.tar.gz](https://github.com/user-attachments/files/15960812/AbeilleLogs-240624.tar.gz)

**Comment by DaddyBabbes** [comment]
et voici les logs inclusion et après avoir taper sur SOS, N°+Cadena fermé, etc.
[AbeilleLogs-240624 (1).tar.gz](https://github.com/user-attachments/files/15960827/AbeilleLogs-240624.1.tar.gz)

**Comment by tcharp38** [comment]
Pour moi à creuser demain. Je vois pourquoi il n'est pas reconnu.
Il y a un bug

    [2024-06-24 22:08:00]   deviceUpdates(Abeille1, 3943, EP=2C, upd={"servClusters":"0000/0003/0020/0500/0001/0B05"})
    [2024-06-24 22:08:00]     'servClusters'='0000/0003/0020/0500/0001/0B05'. Status=identifying
    [2024-06-24 22:08:00]     Updated eq={"ieee":"0015BC00430049C9","macCapa":"80","rxOnWhenIdle":0,"rejoin":"","status":"identifying","time":1719259680,"endPoints":{"01":{"servClusters":"0005/0006"},"2C":{"servClusters":"0000/0003/0020/0500/0001/0B05"}},"mainEp":"","manufId":null,"modelId":null,"location":null,"modelName":"","modelSource":"","modelForced":false,"manufCode":"1015"}
    [2024-06-24 22:08:00]     Requesting manufId/modelId/location from EP 01

Sinon je vois bien des infos remonter maintenant.
Tu dois le voir sur le dashboard.
Donc.. quel est le status pour toi ?

**Comment by DaddyBabbes** [comment]
Effectivement, maintenant je le vois bien les statuts qui change dans le dashboard. Pourtant, je n'ai pas fait différement que les autres fois. Quel est le bug ? 
Je vais faire des tests de scénarios pour voir si tout fonctionne correctement.

**Comment by tcharp38** [comment]
Dans ton cas le bug était surement dans ton firmware de Zigate.

Maintenant il y en a un pour la reconnaissance automatique. La c'est pour ma pomme

**Comment by tcharp38** [comment]
Serais tu partant pour basculer sur la beta ?
J'ai corrigé qqch pour ce defaut d'identification et j'ai besoin de ton device pour confirmer.

**Comment by DaddyBabbes** [comment]
Pourtant pour le firmware, je n'ai pas pu le mettre à jour. Je suis toujours en 3.21. 
Pour la Béta, pourquoi pas. Pour le moment je n'ai pas créer mes scénarios donc je peux réinitialiser le matériel pour tes tests.

**Comment by tcharp38** [comment]
Ha, pas de mise à jour du FW.
C'est une v2 ou v1 ? 
Tu as un FW 004-0321 donc v1 mais tu as mis "PIv2". Ca peut expliquer le fait que tu ne puisses pas programmer le FW.

Quoi qu'il en soit du coup je ne sais pas pourquoi c'est tombé en marche

**Comment by DaddyBabbes** [comment]
La PiZigate est une v1.
et a ton avis faut il que je change PIv2 en 1 ?

**Comment by tcharp38** [comment]
Oui ca c'est sur. C'est incohérent.
Mais ca n'aura d'impact que sur la programmation eventuellement

**Comment by DaddyBabbes** [comment]
Qu'entends tu par programmation ? La mise à jour du Firmware ? Parce que la elle fonctionne bien.

**Comment by tcharp38** [comment]
Oula.. j'ai du mal à te suivre.
Donc tu as pu mettre à jour le FW de la Zigate et tu n'es pus en 3.21. C'est ca ?
Ou elle fonctionne bien avec le vieux FW 3.21 ?

**Comment by DaddyBabbes** [comment]
Lol, non. Je suis toujours en 3.21. Je n'ai pas pu mettre à jour le FW. Quiproquos car tu parles d'impacte éventuel sur la programmation mais je vois pas ce que tu appels programmation. 
Actuellement, elle fonctionne bien avec le 3.21.
Je viens aussi de commander la PiZiGate+ dernièrement que je n'ai pas encore reçu pour mettre à jour mes composants sur un RPi4 au lieu de la vieille PiZigate v1 sur RPi3

**Comment by tcharp38** [comment]
Salut
As tu déja commandé ta pizigate + ?

**Comment by DaddyBabbes** [comment]
Oui, je viens de la recevoir pourquoi ?Le 28 juin 2024 13:10, Tcharp38 ***@***.***> a écrit :
Salut
As tu déja commandé ta pizigate + ?

**Comment by tcharp38** [comment]
Désolé. Je suis trop bête. J'aurais du réagir avant pour te conseiller de reprendre une v1 d'occase, avec antenne. Il y en a plein sur le bon coin.

**Comment by DaddyBabbes** [comment]
Pourquoi ? la v2 n'est pas bien ? 

GOURLAND Kévin 
43, Avenue des Chardonnerets 
77178 SAINT-PATHUS 

[ https://www.poulpeo.com/p/2wqiab |   ] [ https://www.strava.com/athletes/60391779 |   ] 


De: "Tcharp38" ***@***.***> 
À: "KiwiHC16/Abeille" ***@***.***> 
Cc: "kgourland" ***@***.***>, "Mention" ***@***.***> 
Envoyé: Vendredi 28 Juin 2024 14:50:39 
Objet: Re: [KiwiHC16/Abeille] Frient KEYZB-110 (Issue #2715) 





Désolé. Je suis trop bête. J'aurais du réagir avant pour te conseiller de reprendre une v1 d'occase, avec antenne. Il y en a plein sur le bon coin.

**Comment by tcharp38** [comment]
Il y a malheureusement parfois des comportements étranges. Et plus de mises à jour depuis un bail. Je la considère bien moins mur. Ce que j'écris dans la release notes et dans certains sujets v2/+ depuis un moment.

On croise les doigts que n'aies pas de soucis.

**Comment by DaddyBabbes** [comment]
Salut,
Sur mon setup actuel (RPi3+Zigate v1) et le clavier. Sais tu comment gérer les leds ?

**Comment by tcharp38** [comment]
Salut
Les LEDs du clavier ?
A prioris ca n'est pas controlable.
https://www.zigbee2mqtt.io/devices/KEYZB-110.html

**Comment by DaddyBabbes** [comment]
Dans ce post, il semble que ce soit possible en Zigbee
https://community.jeedom.com/t/integration-clavier-alarme-zigbee-frient/84865/32

**Comment by tcharp38** [comment]
Ok reste à comprendre comment.
La j'ai pas du tout le temps avant de m'eclipser 3 semaines.
Je te laisse creuser pour voir si tu comprends l'astuce.
Je reprendrai à mon retour debut aout.

**Comment by DaddyBabbes** [comment]
pas de soucis, je vais creuser de mon côté et passe de bonne vacances.

**Comment by tcharp38** [comment]
Salut

I'm back.
As trouvé des infos ?
Quel genre de controle esperes tu faire avec les LEDS ? Sauf erreur il y en a 3 qui refletent un etat different du clavier.

**Comment by DaddyBabbes** [comment]
Bonjour,
Idem, de retour ce lundi après 3 semaines d'absences, j'ai juste eux le temps de finir mes scénraios gestion d'alarme.
Concernant les LED, voir si ce serait possible :
- En mode Alarm All Zone : Allumer la LED rouge de gauche.
- En mode Disarm : Allumer la LED verte de droite
Laisser la LED centrale pour le mode SOS.
J'ai vu dans une vidéo en anglais avec un autre logiciel que Jeedom quand la personne active son alarme, la LED de gauche passe en rouge et en vert quand le mode est désarmé.
Après je t'avoue que je ne suis pas du tout dans le développement. Juste des notions assez basiques sur Jeedom.

**Comment by tcharp38** [comment]
Salut
Pas encore trouvé l'astuce pour les controller mais ces LEDs n'ont pas déja un fonctionnement par défaut ?
